### PR TITLE
File content provider: if selection and selectionArgs are null, let them pass

### LINKED ISSUE #1842: Unable to Upload file...
### Actual behaviour
-Tell us what happens
When try to Upload a file I give connection Error... 
Whit Google Store app version Upload file work correct... 
### Environment data
Android version:
7.1.2
Device model: 
Sm-t805

Nextcloud app version:
2.0.0 build 20000099, latest commit here 2 day ago
Nextcloud server version:
12.0.4

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix #1842

Otherwise we get an exception

### CODE REVIEW

File: src/main/java/com/owncloud/android/providers/FileContentProvider.java [hunk 1]
```diff
@@ -592,7 +592,8 @@ private Cursor query(
             sqlQuery.setProjectionMap(projectionMap);
         }
 
-        if (selectionArgs == null) {
+        // if both are null, let them pass to query
+        if (selectionArgs == null && selection != null) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/providers/FileContentProvider.java]: If ```selection``` is not ```null``` why would we set it to ```selection = "(?)";``` then? ;)

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/providers/FileContentProvider.java]: When doing it the way with (?) and passing the args to it we get automatically prevention of sql injection.

**Comment by tobiasKaminsky** [comment]
@mario @AndyScherzinger we need this soon in, otherwise dev version will not upload anything.

File: src/main/java/com/owncloud/android/providers/FileContentProvider.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/providers/FileContentProvider.java]: 👍

**Comment by AndyScherzinger** [comment]
👍 fine by me

**Comment by mario** [comment]
👍

**Pull request merged**